modules/ChargerInfo/main/charger_informationImpl.cpp

```diff
@@ -0,0 +1,69 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Pionix GmbH and Contributors to EVerest
+
+#include "charger_informationImpl.hpp"
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+
+namespace module {
+namespace main {
+
+void charger_informationImpl::init() {
+}
+
+void charger_informationImpl::ready() {
+    json info(handle_get_charger_information());
+
+    EVLOG_debug << "ChargerInformation: " << info.dump();
+}
+
+std::string charger_informationImpl::load_fw_version_from_file(const std::string& fn) {
+    std::ifstream fw_version_file(fn);
+    std::string first_line;
+
+    if (!fw_version_file)
+        return "";
+
+    if (!std::getline(fw_version_file, first_line))
+        return "";
+
+    return first_line;
+}
+
+types::charger_information::ChargerInformation charger_informationImpl::handle_get_charger_information() {
+    std::vector<std::string> keys = {"vendor",           "model",     "serial",   "friendly_name", "manufacturer",
+                                     "manufacturer_url", "model_url", "model_no", "revision",      "board_revision",
+                                     "firmware_version"};
+    json info = {};
+
+    // iterate over all linked key-value stores and merge all items,
+    // when a key exists in more than one kvs then the last one wins
+    for (const auto& kvs : mod->r_kvs) {
+        for (const auto k : keys)
+            if (kvs->call_exists(k)) {
+                const auto v = kvs->call_load(k);
+                info[k] = std::get<std::string>(v);
+            }
+    }
```

Comment:
Is your intention to have this information distributed in multiple modules providing the `kvs` interface?
Otherwise storing this as a complete json object may be a bit more efficient (you could still merge this afterwards)

Reply:
The idea is to have different kvs modules which provide different keys and/or same keys but the later wins. Yes, having it as complete object and then merge all together would be an option, but the current kvs interface does not support such kind of access. To be honest, I'm not too worried about performance here, since this is almost static data and I assume that the number of hits is manageable.